projects/hello_adc/src/hello_adc.c

```diff
@@ -0,0 +1,39 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define BUTTON_PIN 178
+#define READ_FROM_PIN 166
+
+GpioItCallback prv_log_pin_data(GpioAddress *button_addr, void *context);
+
+int main() {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  GpioAddress button_addr = { GPIO_PORT_A, BUTTON_PIN };
+  GpioAddress read_addr = { GPIO_PORT_A, READ_FROM_PIN };
+  // Setting pin value is not necessary
+  GpioSettings null_settings = {};
```

Comment:
The GPIO settings aren't just for setting the initial pin value: you'll need to specify the pin direction (should be `GPIO_DIR_IN`) and for any address you're going to do an ADC read on, you'll need to specify the alt function as `GPIO_ALTFN_ANALOG`.